Fix EI to work with 2D `y_data` as used in GEKPLS

Laurentww opened the pull request:
Fix for  PR #340. `EI()` Now correctly identifies the optimal design vector from `y_data`

relf commented:
Thanks for the fix. Would you add a test with EI to check it works (regression test for further developments)?

Laurentww commented:
Added the test. 

Also made the following two other code changes, which simplify testing of EI. Let me know if these need a separate PR or are not desired.
 - Removed the `y_data` argument `ego.EI()` . `y_data` now is read from the gpr, which contains the same `y-data`
 - Refactored training of gpr into private ego method

Laurentww commented:
Updated the PR. Thanks for suggesting the assertion fix!
